### PR TITLE
docs: add guidance on extension metadata

### LINKED ISSUE #1057: docs: update UI ext docs to detail metadata requirements
With changes to how we load UI extensions, it becomes essential that plugins define their extensions with the plugin.json (see https://grafana.com/developers/plugin-tools/reference/plugin-json#extensions) 

```[tasklist]
### Tasks
- [x] Update how to guides
- [x] Review Metadata (plugin.json)
```

### PR DESCRIPTION
**Pull request by sympatheticmoose**
**What this PR does / why we need it**:

Clarifies the metadata description for Extensions (note a follow up PR is required in grafana/grafana assuming the wording is approved here).

Adds instructions within the How to guide to advise on this requirement.

**Which issue(s) this PR fixes**:

Fixes #1057 

**Special notes for your reviewer**:

### CODE REVIEW

File: docusaurus/docs/how-to-guides/ui-extensions/register-an-extension.md [hunk 1]
```diff
@@ -21,6 +21,12 @@ Read more about extensions under [key concepts](../../key-concepts/ui-extensions
 
 <ExtensionPoints/>
 
+:::warning
+
+You must [update your plugin.json metadata](#updating-pluginjson-metadata) to list any registered extensions.
```
Comment (josmperez) [inline docusaurus/docs/how-to-guides/ui-extensions/register-an-extension.md:26]:
```suggestion
You must [update your `plugin.json` metadata](#updating-pluginjson-metadata) to list any registered extensions.
```

File: docusaurus/docs/how-to-guides/ui-extensions/register-an-extension.md [hunk 2]
```diff
@@ -217,6 +223,25 @@ export const plugin = new AppPlugin().addComponent({
 });
 ```
 
+## Updating plugin.json metadata
+
+Once you have defined a link or component extension to be registered against an extension point, you must update your plugin.json metadata.
```
Comment (josmperez) [inline docusaurus/docs/how-to-guides/ui-extensions/register-an-extension.md:228]:
```suggestion
Once you have defined a link or component extension to be registered against an extension point, you must update your `plugin.json` metadata.
```

File: docusaurus/docs/how-to-guides/ui-extensions/register-an-extension.md [hunk 3]
```diff
@@ -217,6 +223,25 @@ export const plugin = new AppPlugin().addComponent({
...
+    {
+      "extensionPointId": "grafana/dashboard/panel/menu/v1",
+      "type": "link"
+      "title": "My app",
+      "description": "Link to my app"
+    }
+ ]
+```
+
+For more information, see the plugin.json [reference](../../reference/metadata.md#extensions).
```
Comment (josmperez) [inline docusaurus/docs/how-to-guides/ui-extensions/register-an-extension.md:243]:
```suggestion
For more information, see the `plugin.json` [reference](../../reference/metadata.md#extensions).
```

**Review by josmperez — APPROVED** [review]
LGTM.

File: docusaurus/docs/how-to-guides/ui-extensions/register-an-extension.md (hunk 1 above)
Comment (sympatheticmoose) [reply docusaurus/docs/how-to-guides/ui-extensions/register-an-extension.md:26]: mixing monospace style in a hyperlink looks awful so I'll rework this instead

**Review by sunker — APPROVED** [review]
🚀

**Pull request merged**